qutebrowser/keyinput/modeparsers.py

```diff
@@ -189,7 +189,7 @@ def _handle_special_key(self, e):
                 return True
             else:
                 return super()._handle_special_key(e)
-        elif config.get('hints', 'mode') != 'number':
+        elif hintmanager.current_mode() != 'number':
```

Comment:
I disagree with both :wink: Doing `hintmanager._context.hint_mode` would mean accessing a "private" attribute (and making the whole context public isn't nice either).

As for the latter, the whole point of this PR is adding `--mode` to `:hint` to override the mode on a case-by-case basis :wink: